Edge/edge on kubernetes (#1399)
SoftwareAG/c8y-docs: content/edge-k8s/installing-edge-on-k8-bundle/install-edge.md

---
weight: 20
title: Installing Cumulocity IoT Edge
layout: redirect
---

To install {{< product-c8y-iot >}} Edge, create a Kubernetes manifest file with an Edge CR that describes {{< product-c8y-iot >}} Edge. Use `kubectl` to apply the Edge CR to your Kubernetes cluster.

Run the command below to install {{< product-c8y-iot >}} Edge after you’ve made the necessary updates to the Edge CR ([c8yedge.yaml](/files/edge-k8s/c8yedge.yaml)), which includes the Edge CR and the required secrets:

```bash
kubectl apply -f c8yedge.yaml
```
For more information about the structure and configuration options available in the Edge CR, see the [Edge Custom Resource](/edge-k8s/edge-custom-resource-definition/) section.

> ### Verify Edge installation

Reviewer: ```suggestion
### Verify the Edge installation
```

To monitor the installation progress, run the command below: 

```shell
kubectl describe edge c8yedge -n c8yedge
```

> This command allows you to view the details about the installation of **c8yedge** in the **c8yedge** namespace. 

Reviewer: ```suggestion
This command allows you to view the details about the installation of *c8yedge* in the *c8yedge* namespace. 
```

{{< c8y-admon-info >}}

> Substitute the Edge name and namespace name, which is currently “c8yedge” in the command, with the specific Edge name and namespace name you've specified in your Edge CR. 

Reviewer: ```suggestion
Substitute the Edge name and namespace name, which is currently *c8yedge* in the command, with the specific Edge name and namespace name you have specified in your Edge CR. 
```

{{< /c8y-admon-info >}}

You can also follow the events raised for the Edge CR by running the command below: 

```shell
kubectl get events -n c8yedge –field-selector involvedObject.name=c8yedge –watch 
```

> The **Events** section in the output of the `describe edge` command specifies the installation progress and the **Status** section displays the generation of the Edge CR which is being installed and its current state. Once the installation succeeds, the **Status** section also displays the generation of the CR which is deployed, version of the Edge, last deployed time/age, validation warnings, if any and some help commands for downloading the diagnostic logs, extracting the Root CA of the Operator generated TLS certificates.

Reviewer: ```suggestion
The **Events** section in the output of the `describe edge` command specifies the installation progress and the **Status** section displays the generation of the Edge CR which is being installed and its current state. Once the installation succeeds, the **Status** section also displays the generation of the CR which is deployed, version of the Edge, last deployed time/age, validation warnings, if any and some help commands for downloading the diagnostic logs, extracting the Root CA of the operator generated TLS certificates.
```

Author: This has to be in uppercase.

> The **Events** section in the output of the `describe edge` command specifies the installation progress and the **Status** section displays the generation of the Edge CR which is being installed and its current state. Once the installation succeeds, the **Status** section also displays the generation of the CR which is deployed, version of the Edge, last deployed time/age, validation warnings, if any and some help commands for downloading the diagnostic logs, extracting the Root CA of the Operator generated TLS certificates.

Reviewer: "version of the Edge" -> "{{< product-c8y-iot >}} Edge version" ?
